eng/performance/benchmark_jobs.yml

```diff
@@ -76,20 +76,23 @@ jobs:
         steps:
         - checkout: self
           clean: true
+        - ${{ if eq(parameters.osName, 'windows') }}:
+          - script: (robocopy $(Build.SourcesDirectory) $(Build.SourcesDirectory)\notLocked /E /XD $(Build.SourcesDirectory)\notLocked $(Build.SourcesDirectory)\artifacts $(Build.SourcesDirectory)\.git) ^& IF %ERRORLEVEL% LEQ 1 exit 0
+            displayName: Workaround to avoid locked file # https://github.com/dotnet/arcade/issues/2125
         - template: /eng/common/templates/steps/send-to-helix.yml
           parameters:
             HelixSource: 'pr/dotnet/performance/$(Build.SourceBranch)' # sources must start with pr/, official/, prodcon/, or agent/
             HelixType: 'test/performance_$(_BuildConfig)/'
             HelixAccessToken: $(HelixApiAccessToken)
             HelixTargetQueues: ${{ parameters.queue }}
-            HelixPreCommands: $(HelixPreCommand)
             Creator: $(Creator)
             WorkItemTimeout: 4:00 # 4 hours
             ${{ if eq(parameters.osName, 'windows') }}:
-              WorkItemDirectory: '$(Build.SourcesDirectory)\src\benchmarks\micro'
-              WorkItemCommand: 'py -3 %HELIX_CORRELATION_PAYLOAD%\benchmarks_ci.py --bin-directory ".\bin" --working-directory . --incremental no --category $(_Category) --architecture ${{ parameters.architecture }} -f $(_Framework) $(BenchmarkDotNetArguments) $(BenchViewArguments)'
-              CorrelationPayloadDirectory: '$(Build.SourcesDirectory)\scripts'
+              HelixPreCommands: '$(HelixPreCommand);reg export HKLM\SYSTEM\CurrentControlSet\Control\FileSystem backup.reg;reg add HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\FileSystem /v LongPathsEnabled /t REG_DWORD /d 1 /f'
```

Comment:
Please don't modify machine state. This will change it for everything that comes after it.

Reply:
But I restore the settings in the Post command. Is it not enough?

Reply:
Will your tool absolutely never crash and are you ensuring the power does not go out before your post runs?

Reply:
Let me talk to some people. Generally, this isn't something we really want to do simply, at least to me, in the instance that something goes wrong and, say, the post command doesn't get run. I'm not sure if that's a situation we could run into, I just don't want us to run the risk.

Reply:
Where are we running into the max-path issue, and can we be smarter about our filenames such that we don't?

Reply:
The first time is when the python script calls the powershell script to install dotnet cli:

```log
DotNet Install Path: 'C:\dotnetbuild\work\caadb38f-4fcf-4180-968e-fdfd3776b696\Work\2757c072-fe57-4477-8760-154f8f4d9423\Exec\tools\dotnet\x64'
Downloading https://dot.net/v1/dotnet-install.ps1
$ pushd "C:\dotnetbuild\work\caadb38f-4fcf-4180-968e-fdfd3776b696\Work\2757c072-fe57-4477-8760-154f8f4d9423\Exec"
$ powershell.exe -NoProfile -ExecutionPolicy Bypass C:\dotnetbuild\work\caadb38f-4fcf-4180-968e-fdfd3776b696\Work\2757c072-fe57-4477-8760-154f8f4d9423\Exec\tools\dotnet\x64\dotnet-install.ps1 -InstallDir C:\dotnetbuild\work\caadb38f-4fcf-4180-968e-fdfd3776b696\Work\2757c072-fe57-4477-8760-154f8f4d9423\Exec\tools\dotnet\x64 -Architecture x64 -Channel master
dotnet-install: Downloading link: https://dotnetcli.azureedge.net/dotnet/Sdk/3.0.100-preview4-010525/dotnet-sdk-3.0.100-preview4-010525-win-x64.zip
dotnet-install: Extracting zip from https://dotnetcli.azureedge.net/dotnet/Sdk/3.0.100-preview4-010525/dotnet-sdk-3.0.100-preview4-010525-win-x64.zip
C:\dotnetbuild\work\caadb38f-4fcf-4180-968e-fdfd3776b696\Work\2757c072-fe57-4477-8760-154f8f4d9423\Exec\tools\dotnet\x64\dotnet-install.ps1 : Exception calling "ExtractToFile" with "3" argument(s): "Could not find a part of the path 'C:\dotnetbuild\work\caadb38f-4fcf-4180-968e-fdfd3776b696\Work\2757c072-fe57-4477-8760-154f8f4d9423\Exec\tools\dotnet\x64\packs\Microsoft.AspNetCore.App.Ref\3.0.0-preview4-19123-01\ref\netcoreapp3.0\Microsoft.AspNetCore.Server.Kestrel.Transport.Abstractions.dll'."
```

Reply:
Also here I am doing this only for Windows and we have dedicated Perf Windows queues. Do machines from the perf queues belong to other queues?

Windows.Amd64.ClientRS4.DevEx.15.8.Perf
Windows.10.Amd64.ClientRS1.Perf

Reply:
Ok, I was able to work around it by using `CorrelationPayloadDirectory` instead of `WorkItemDirectory`

the difference is the length of the path of the folder where they get checked out in Helix...

```log
C:\dotnetbuild\work\82ac7b90-beba-445c-96a0-c3ed279fccb2\Work\e11567ec-0440-4e29-b6c0-e1f673b64baf\Exec - WorkItemDirectory
C:\dotnetbuild\work\82ac7b90-beba-445c-96a0-c3ed279fccb2\Payload\ - CorrelationPayloadDirectory
```